toptal/picasso
# style(Modal): fix modal content overflow issue

[SPT-1708](https://toptal-core.atlassian.net/browse/SPT-1708)
[SPT-1712](https://toptal-core.atlassian.net/browse/SPT-1712)

### Description
As `modalContent` is the last element before `children` we should add padding to this element instead of `wrapper`
https://github.com/toptal/picasso/blob/1be0c8d2075a3b7cae6f3681da85d05e93e60f37/packages/picasso/src/ModalContent/ModalContent.tsx#L34-L43
It will solve following overflow issues:
- SPT-1708: extra scrollbar for negative margins within `children`
- SPT-1712: clipped boxShadow while focusing input

### How to test
1. [https://picasso.toptal.net/SPT-1712-SPT-1708-fix-modal-content-overflow-issue/](https://picasso.toptal.net/SPT-1712-SPT-1708-fix-modal-content-overflow-issue/?9f6a73d189c89cae856be7ffddfe92136e91b40b)
2. Go to `Modal & PromptModal` stories.
3. check all examples

### Screenshots SPT-1708 (extra scrollbar for negative margins within `children`)
| Before.                                 | After.                                  |
| --------------------------------------- | --------------------------------------- |
| <img src="https://user-images.githubusercontent.com/4816942/122007675-63775500-cde2-11eb-80ef-8cd27756a742.png" width="600" /> | <img src="https://user-images.githubusercontent.com/4816942/122007680-65d9af00-cde2-11eb-88eb-f7a63987139b.png" width="600" /> |

### Screenshots SPT-1712 (clipped boxShadow while focusing input)
| Before.                                 | After.                                  |
| --------------------------------------- | --------------------------------------- |
| <img src="https://user-images.githubusercontent.com/4816942/122008020-c4069200-cde2-11eb-988a-b371a03dec2b.png" width="600" /> | <img src="https://user-images.githubusercontent.com/4816942/122008105-d5e83500-cde2-11eb-8022-d2274897cc3b.png" width="600" /> |

### Review

<details>
<summary>PR commands</summary>
<br />

List of available commands:

- `@toptal-bot run all` - Run whole pipeline
- `@toptal-bot run build` - Check build
- `@toptal-bot run visual` - Run visual tests
- `@toptal-bot run deploy:documentation` - Deploy documentation
- `@toptal-bot run package:alpha-release` - Release alpha version

</details>
